src/EmailSynchronizer.php

```diff
@@ -137,24 +137,30 @@ public function synchronizeEmail(int $number, string $source): void
             }
         }
         // Extract the thread ID from the "references" header
-        $threadId = null;
+        $firstReference = null;
         $references = $parsedDocument->getHeaderValue('References');
         if ($references) {
             $references = preg_split('/(?<=>)/', $references);
             $references = array_filter(array_map('trim', $references));
             if (! empty($references)) {
-                $threadId = reset($references);
+                $firstReference = reset($references);
                 if (! $inReplyTo) {
                     // In old mails the `In-Reply-To` header didn't exist, instead it was at the end of the references
                     // Example: https://externals.io/message/2536#2784
                     $inReplyTo = end($references);
                 }
             }
         }
-        // We know it is a reply to an email but we weren't able to find the thread ID: let's find it from our database
-        if ($threadId === null && $inReplyTo !== null) {
+
+        $threadId = null;
+        if ($firstReference !== null) {
```

Comment:
The processing around this can be written in one line, but it is written this way for readability.